lib/benchpark/experiment.py

```diff
@@ -48,10 +49,153 @@ class Experiment(ExperimentSystemBase):
+    )
+
     def __init__(self, spec):
         self.spec: "benchpark.spec.ConcreteExperimentSpec" = spec
         super().__init__()
 
+    # input parameters:
+    # 1. input_variables: dictionary with key value pairs of type str: int or tuple(str): list(int)
+    # For the value in input_variables corresponding to scaling_variable,
+    # if the value is a list, select the index of its smallest element, 0 otherwise
+    # Beginning with this index, generate a list of indexes of length equal to
+    # the number of dimensions in an (ascending) round-robin order
+    # 2. scaling_variable: variable of type str or tuple(str). The scaling order is determined by
+    # the value in input_variables corresponding to scaling_variable.
+    #
+    # output:
+    # scaling_order: list[int]. list of indices, with one value for each dimension,
+    # starting with the minimum value of the first element in input_variables arranged
+    # in an ascending round-robin order
+    def configure_scaling_policy(self, input_variables, scaling_variable):
+        # compute the number of dimensions
+        n_dims = 1
+        for param in input_variables.values():
+            if isinstance(param, list):
+                n_dims = len(param)
+                break
+
+        # starting with the minimum value dim of the scaling_variable
+        # compute the remaining n_dims-1 values in a round-robin manner
+        val = input_variables[scaling_variable]
+        min_dim = val.index(min(val)) if isinstance(val, list) else 0
+
+        return [(min_dim + i) % n_dims for i in range(n_dims)]
+
+    # input parameters:
+    # 1. input_variables: dict[str, int | tuple(str), list[int]]. Dictionary of all variables
+    # that need to be scaled. All variables are ordered as per the ordering policy of
+    # the first element in input_variables. By default, this policy is to scale the
+    # values beginning with the smallest dimension and proceeding in a RR manner through
+    # the other dimensions
+    #
+    # 2. scaling_factor: int. Factor by which to scale the variables. All entries in
+    # input_variables are scaled by the same factor
+    #
+    # 3. num_exprs: int. Number of experiments to be generated
+    #
+    # 4. scaling_variable: variable of type str or tuple(str). The scaling order is determined by
+    # the value in input_variables corresponding to scaling_variable. If no scaling_variable is
+    # specified, the scaling order is defined using the first element in input_variables
+    #
+    # output:
+    # scaling_order: list[int]. list of indices, with one value for each dimension,
+    # output:
+    # output_variables: dict[str, int | list[int]]. num_exprs values for each
+    # dimension of the input variable scaled by the scaling_factor according to the
+    # scaling policy
+    def scale_experiment_variables(
+        self, input_variables, scaling_factor, num_exprs, scaling_variable=None
```

Comment:
Hi @dyokelson, I've added the option to specify a scaling_variable to the experiment (all variables in `input_variables` will be scaled according to the ordering defined on `scaling_variable`